packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -35,13 +50,55 @@
       limit: 100,
     },
   })
+  $: hasFilter = !!filter?.filter(f => !!f.field)?.length
   $: fetch.update({ filter })
-  $: options = $fetch.rows
+  $: {
+    options = searchResults ? searchResults : $fetch.rows
+    const nonMatchingOptions = selectedOptions.filter(
+      option => !options.map(opt => opt._id).includes(option._id)
+    )
+    // Append initially selected options if there is no filter
+    // and hasn't already been appended
+    if (!hasFilter) {
+      options = [...options, ...nonMatchingOptions]
+    }
+  }
   $: tableDefinition = $fetch.definition
+  $: primaryDisplay = tableDefinition?.primaryDisplay || "_id"
   $: singleValue = flatten(fieldState?.value)?.[0]
   $: multiValue = flatten(fieldState?.value) ?? []
   $: component = multiselect ? CoreMultiselect : CoreSelect
   $: expandedDefaultValue = expand(defaultValue)
+  $: debouncedSearch(searchString)
+  $: {
+    if (searching) {
+      isOpen = true
+    }
+  }
+
+  $: {
+    if (
+      primaryDisplay !== "_id" &&
+      fieldState?.value?.length &&
+      !selectedOptions?.length
+    ) {
+      API.searchTable({
+        paginate: false,
+        tableId: linkedTableId,
+        limit: 100,
+        query: {
+          oneOf: {
+            [`1:${primaryDisplay}`]: fieldState?.value?.map(
+              value => value.primaryDisplay
+            ),
+          },
+        },
+      }).then(response => {
```

Comment:
I'm not sure what the purpose of this is... I can see that you're searching the table to retrieve the full rows for each of the selected option, but I don't know why that's needed. The data we already have in the initial row includes the primary display of each option, so we have all the data we need to be able to render the selected values.

Perhaps the bigger limitation here is the fact that it's a multi select and depends upon having "options" available for each selected value.

Reply:
This is required to fix the bug were the selected option(s) were null if not within the first 100 records. 

It also ensures that the options list include the initially selected options so they can be deselected without having to search for them, as demonstrated in the video.